ServiceWebsite/ServiceWebsite/Controllers/ParticipantController.cs

```diff
@@ -34,8 +39,63 @@ public ParticipantController()
         [ProducesResponseType((int)HttpStatusCode.BadRequest)]
         public async Task<IActionResult> UpdateSuitabilityAnswers(Guid hearingId, [FromBody]List<HearingSuitabilityAnswer> answers)
         {
-          await Task.Yield();
-          return NoContent();
+            if (hearingId == Guid.Empty)
+            {
+                return new BadRequestResult();
+            }
+            try
+            {
+                var participantId = await _hearingService.GetParticipantId(User.Identity.Name, hearingId);
+                
+                if (answers.Count == 0)
+                {
+                    return new BadRequestResult();
+                }
+                else
+                {
+                    foreach (HearingSuitabilityAnswer answer in answers)
+                    {
+                        bool isValid = RegExValidators.ValidateForCapsAndUnderscore(answer.QuestionKey);
+                        if (!isValid)
+                        {
+                            return new BadRequestResult();
+                        }
+                    }
+                }
+
+                var suitabilityAnswers = MapAnswers(answers);
+                await _participantService.UpdateSuitabilityAnswers(hearingId, participantId, suitabilityAnswers);
+
+                return NoContent();
+            }
+            catch (NotFoundException e)
+            {
+                ApplicationLogger.TraceException(TraceCategories.MissingResource, "Missing hearing for user", e, User);
+                return NotFound($"No hearing with id '{hearingId}' found for user");
```

Comment:
We don't know if it is the hearing that is missing do we? Maybe it's better to return it with the same error message?

Reply:
Changed